lib/wallet/http.js

```diff
@@ -833,6 +783,405 @@ class HTTP extends Server {
       await req.wallet.resend();
       res.json(200, { success: true });
     });
+
+    // Wallet Name
+    this.get('/wallet/:id/name', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const name = valid.str('name');
+
+      assert(name, 'Must pass name.');
+      assert(rules.verifyName(name), 'Must pass valid name.');
+
+      const height = this.wdb.height;
+      const network = this.network;
+      const ns = await req.wallet.getNameStateByName(name);
+
+      res.json(200, ns.getJSON(height, network));
+    });
+
+    // Wallet Names/Auctions
+    this.get('/wallet/:id/names', async (req, res) => {
+      const height = this.wdb.height;
+      const network = this.network;
+
+      const names = await req.wallet.getNames();
+      const items = [];
+
+      for (const ns of names)
+        items.push(ns.getJSON(height, network));
+
+      res.json(200, items);
+    });
+
+    // Wallet Auction
+    this.get('/wallet/:id/auction', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const name = valid.str('name');
+
+      const height = this.wdb.height;
+      const network = this.network;
+
+      assert(name, 'Must pass name.');
+      assert(rules.verifyName(name), 'Must pass valid name.');
+
+      const ns = await req.wallet.getNameStateByName(name);
```

Comment:
Should this throw if name isn't found in wallet?  I tried breaking your test:
`Error: Cannot read property 'getJSON' of null`

Reply:
Good catch here - there needs to be a test case for this, will add one

Reply:
Added a testcase for this here https://github.com/handshake-org/hsd/pull/163/files#diff-7b92af4a1865adf03377c0804427d57dR535